configure

```diff
@@ -6482,6 +6482,14 @@ if test "$Py_OPT" = 'true' ; then
   DEF_MAKE_ALL_RULE="profile-opt"
   REQUIRE_PGO="yes"
   DEF_MAKE_RULE="build_all"
+  case $CC in
```

Comment:
There is no conditional activation on `--enable-shared` because C extensions can also benefit from this (and they inherit the build flags).